test/python/algorithms/test_estimator_gradient.py

```diff
@@ -0,0 +1,356 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# =============================================================================
+
+""" Test Quantum Gradient Framework """
+
+import unittest
+from test import combine
+
+import numpy as np
+from ddt import ddt
+
+from qiskit import QuantumCircuit
+from qiskit.algorithms.gradients import (
+    FiniteDiffEstimatorGradient,
+    LinCombEstimatorGradient,
+    ParamShiftEstimatorGradient,
+    SPSAEstimatorGradient,
+)
+from qiskit.circuit import Parameter
+from qiskit.circuit.library import EfficientSU2, RealAmplitudes
+from qiskit.circuit.library.standard_gates import RXXGate, RYYGate, RZXGate, RZZGate
+from qiskit.primitives import Estimator, Sampler
+from qiskit.quantum_info import SparsePauliOp
+from qiskit.quantum_info.random import random_pauli_list
+from qiskit.test import QiskitTestCase
+
+
+@ddt
+class TestEstimatorGradient(QiskitTestCase):
+    """Test Estimator Gradient"""
+
+    @combine(
+        grad=[FiniteDiffEstimatorGradient, ParamShiftEstimatorGradient, LinCombEstimatorGradient]
+    )
+    def test_gradient_p(self, grad):
+        """Test the estimator gradient for p"""
+        estimator = Estimator()
+        a = Parameter("a")
+        qc = QuantumCircuit(1)
+        qc.h(0)
+        qc.p(a, 0)
+        qc.h(0)
+        if grad is FiniteDiffEstimatorGradient:
+            gradient = grad(estimator, epsilon=1e-6)
+        else:
+            gradient = grad(estimator)
+        op = SparsePauliOp.from_list([("Z", 1)])
```

Comment:
It would be good to test against a `PauliSumOp` and other `BaseOperator`s too, like e.g. `Operator` 🙂 
```suggestion
        op = PauliSumOp.from_list([("Z", 1)])
```